spine-ts/spine-player/src/Player.ts

```diff
@@ -551,6 +582,63 @@ export class SpinePlayer implements Disposable {
 		}
 	}
 
+	private setUpAudio() {
+		this.animationState!.addListener({
+			event: (entry, event) => {
+				if (this.audioEnabled) {
+					const audioPath = event.data.audioPath
+					if (audioPath) {
+						const playAudio = this.audioNodes[`${audioPath}-${event.balance}-${event.volume}`];
+						if (playAudio) 
+							playAudio();
+						if (!playAudio) {
+							const audioNodes = [] as AudioNode[];
+							if (typeof this.audioCtx!.createGain === 'function') {
+								const node = this.audioCtx!.createGain();
+								node.gain.setValueAtTime(event.volume, this.audioCtx!.currentTime);
+								audioNodes.push(node);
+							}
+
+							// Cannot use createStereoPanner because some browser does not support it yet
+							if (typeof this.audioCtx!.createPanner === 'function') {
+								const node = this.audioCtx!.createPanner();
+								if (node.positionX !== undefined) {
+									node.positionX.value = event.balance;
+								} else {
+									node.setPosition(event.balance, 0, 1 - Math.abs(event.balance));
+									node.panningModel = 'equalpower';
+								}
+								audioNodes.push(node);
+							}
+
+							audioNodes.push(this.audioCtx!.destination);
+
+							// Need to clone the ArrayBuffer since 'decodeAudioData' take Buffer ownership forever
+							const soundSrc = this.assetManager!.require([this.config.audioFolderUrl, audioPath].join("/"));
+
+							this.audioCtx!.decodeAudioData(soundSrc.slice(0),
```

Comment:
This means we fetch the `ArrayBuffer` representing the audio from the asset manager, and decode it every time an audio event is encountered. That doesn't seem great. Can we decode the audio once and store that in asset manager instead of the undecoded `ArrayBuffer`? The constructor of `AssetManagerBase` can get an optional parameter like `audioLoader`, which should be called by AssetManager when the audio `ArrayBuffer` is ready. The `audioLoader` can then do whatever, like decode the `ArrayBuffer` to an `AudioBuffer`. It will be the `AudioBuffer` that is stored in and returned by the asset manager, not the raw undecoded `ArrayBuffer`. Then the code below can simply fetch the `AudioBuffer` from the asset manager on demand and we only create new buffer sources instead of decoding the audio over and over again.